src/linear.jl

```diff
@@ -0,0 +1,24 @@
+import ChainRulesCore: rrule
+using LuxCore
+using Random
+
+struct LinearLayer <: AbstractExplicitLayer 
+   in_dim::Integer
+   out_dim::Integer
+end
+ 
+function (l::LinearLayer)(x::AbstractMatrix, ps, st)
+   return ps.W * parent(x), st
```

Comment:
Isn't this behaviour incorrect? When x is a matrix, then the leading dimension is the batch-index, and the row-index is the output-index from the previous layer. So the logical operation would actually be 
```julia
parent(x) * transpose(ps.W)
```

Reply:
I appreciate this is confusing regardless how we decide. Needs some thought and maybe another strong argument to introduce the `Batch` type.

Reply:
For this implementation I tried to follow what `Lux` is doing exactly, but I agree with what you said. How about introducing a `feature_first` argument so that the user can choose what they want?  This is also the way that `conv2d` in `pytorch` solves a similar problem.

Reply:
let's take this to the main discussion thread. Can you explain there, what this would look lie?